gwu-libraries/scholarspace-hyrax
# T391-2-rspec-setup

# PR 2 of 3 for automated test and CI/CD

Marking this as a draft PR until the docker/env changes in PR #477 are confirmed and merged. 

This PR is focused on the configuration of testing suite tools, including:
- RSpec
- FactoryBot
- SimpleCov
- DatabaseCleaner

## To test:

- Build and start Docker containers
- Start interactive terminal in the app-server container, e.g. `docker exec -it scholarspace-hyrax_app-server_1 /bin/sh`
- Change to `scholarspace` user with `su scholarspace`
- Create dev/test databases with `bundle exec rails db:create`
- Migrate dev/test databases with `bundle exec rails db:migrate`
- Run `bundle exec rspec`

You should see 2 failing tests and 12 pending tests, and no passing tests. PRs for the rspec tests configured for this setup will follow this one. 

![Screen Shot 2024-01-10 at 2 09 15 PM](https://github.com/gwu-libraries/scholarspace-hyrax/assets/17027357/1eb062b4-89b6-4359-9a59-4afc11542447)
